src/route/page.rs

```diff
@@ -18,37 +18,134 @@
  */
 
-use super::prelude::*;
+use crate::{remote::DeepwellPool, route::prelude::*};
+use actix_web::{Error, *};
+use deepwell_core::prelude::*;
+use ftml::{
+    data::User as FtmlUser,
+    handle::{NullHandle, RemoteHandle},
+    prefilter,
+    prelude::*,
+    RemoteError, RemoteResult,
+};
 use std::collections::HashMap;
 use wikidot_path::Request as PageRequest;
 
 // Public route methods
 
+/* TODO: Temporary Measure
+ *
+ * Until we can figure out a good way of divining the WikiId from the wiki request,
+ * just assume that these functions are retrieving the wiki at WikiId 1.
+ */
+lazy_static! {
+    static ref TEMP_WIKI_ID: WikiId = WikiId::from_raw(1);
+}
+
+/// Return an HttpResult for getting a page by its slug, using deepwell.
+pub async fn get_deepwell_page(
+    wiki_id: WikiId,
+    slug: &str,
+    deepwell: &web::Data<DeepwellPool>,
+) -> std::result::Result<HttpResponse, ()> {
+    debug!("Retrieving page by WikiId {} and slug {}", wiki_id, slug);
+    let result = deepwell
+        .claim()
+        .await
+        .get_page_contents(wiki_id.clone(), slug.to_string())
+        .await;
+
+    match try_io_result!(result) {
+        Ok(Some(page)) => {
+            // now run FTML on it
+            let mut contents = match String::from_utf8(page.to_vec()) {
+                Ok(v) => v,
+                Err(e) => return Ok(HttpResponse::InternalServerError().json(format!("{:?}", e))),
+            };
+
+            // TODO: something that uses Deepwell
+            let remote_handle = NullHandle;
+            let renderer = HtmlRender::new(&remote_handle);
+
+            // TODO: also get page metadata
+            let page_info = PageInfo {
+                title: "SCP-XXXX",
+                alt_title: Some("The Monster"),
+                header: None,
+                subheader: None,
+                rating: 1000,
+                tags: vec![
+                    "scp",
+                    "keter",
+                    "intangible",
+                    "k-class-scenario",
+                    "ontokinetic",
+                ],
+            };
+
+            let mut output = match renderer.transform(&mut contents, page_info, &remote_handle) {
+                Ok(o) => o,
+                Err(e) => return Ok(HttpResponse::InternalServerError().json(format!("{:?}", e))),
+            };
+
+            // TODO: use jinja/other templater to put this text into a better template
+            let mut buffer = String::from("<html><head>");
+
+            for meta in &output.meta {
+                meta.render(&mut buffer);
+            }
+
+            buffer.push_str("<style>");
+            buffer.push_str(&output.style);
+            buffer.push_str("</style></head><body>");
+            buffer.push_str(&output.html);
+            buffer.push_str("</body></html>\n");
+
+            Ok(HttpResponse::Ok().json(buffer))
+        }
+        Ok(None) => Err(()),
```

Comment:
Why are we returning an empty error? This case is if there's no page, we should return an `Option` not `Err(())`, it's confusing with the fact that there's an actual error case being handled here.

imo it should return `Option<HttpResponse>`, where `None` means no such page.